instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/internal/InternalNetClientAttributesExtractor.java

```diff
@@ -49,13 +50,14 @@ public void onEnd(AttributesBuilder attributes, REQUEST request, @Nullable RESPO
 
     internalSet(
         attributes, SemanticAttributes.NET_TRANSPORT, getter.getTransport(request, response));
+    String protocolName = getter.getProtocolName(request, response);
+    if (protocolName != null) {
+      internalSet(
+          attributes, NetAttributes.NET_PROTOCOL_NAME, protocolName.toLowerCase(Locale.ROOT));
```

Comment:
what do you think of moving `toLowerCase` into the `getProtocolName` implementations that need it?

Reply:
That was my initial idea, but I've decided to have the conversion here because of how this attribute is described in the spec:

> Application layer protocol used. The value SHOULD be normalized to lowercase.

I'm thinking that any normalization should probably be done in the extractor; there's a similarity to the SQL sanitization, which also happens in the extractor.

Reply:
> Application layer protocol used. The value SHOULD be normalized to lowercase.

I think of the semantic convention definitions as applying to the getter methods, and I think it could become slightly more confusing if some parts of the semantic convention are handled in the extractor and some parts are handled in the getter methods.

I think the existing SQL sanitization is a bit ok, because in that case the getter method is `getRawStatement()` as opposed to `getStatement()`

Reply:
> I think of the semantic convention definitions as applying to the getter methods, and I think it could become slightly more confusing if some parts of the semantic convention are handled in the extractor and some parts are handled in the getter methods.

That's a good point.
My idea about the extractors and semantic conventions was: put as much as possible into extractors, and make implementing the getters as easy as possible - someone who uses the instrumentation api to instrument their own library should not have to know the intricacies of our semantic conventions, the extractors are kinda "smart" and can massage the data a bit. We already do things like that with e.g. removing auth info from `http.url`, or the conditional on the `net.sock.family` (skip if default `inet`).